src/mlpack/methods/ann/activation_functions/elliot_function.hpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file elliot_function.hpp
+ * @author Bisakh Mondal
+ *
+ * Definition and implementation of the Elliot Activation function as described 
+ * by D.L. Elliott
+ *
+ * For more information, see the following paper.
+ *
+ * @code
+ * @techreport{Elliott1993,
+ *   title = {A better activation function for artificial neural networks},
+ *   author = {Elliott, David L},
+ *   url = {https://drum.lib.umd.edu/bitstream/handle/1903/5355/TR\_93-8.pdf}
+ *   year = {1993}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_ELLIOT_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_ELLIOT_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The Elliot function, defined by
+ *
+ * @f{eqnarray*}{
+ *  f(x) &=& \frac{x}{1 + |x|} \\
+ *  f'(x) &=& \frac{1}{(1 + |x|)^2}
+ * @f}
+ */
+class ElliotFunction
+{
+ public:
+  /**
+   * Computes the Elliot function.
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  static double Fn(const double x)
+  {
+    return x / (1.0 + std::abs(x));
+  }
+
+  /**
+   * Computes the Elliot function.
+   *
+   * @param x Input data.
+   * @param y The resulting output activation.
+   */
+  template <typename InputVecType, typename OutputVecType>
+  static void Fn(const InputVecType &x, OutputVecType &y)
+  {
+      y = x / (1.0 + arma::abs(x));
+  }
+
+  /**
+   * Computes the first derivative of the Elliot function.
+   *
+   * @param y Input data.
+   * @return f'(x)
+   */
+  static double Deriv(const double y)
+  {
+    return 1.0 / std::pow(1.0 + std::abs(y), 2);
```

Comment:
I mentioned this problem in the elish function PR you made too, but I thought I should mention it just in case you forget to change it that y is the input here and not x.

Reply:
Hi @prince776
I think variable y is already treated as the input here instead of x, same is said in the comments for the deriv function. If there is something else you pointing out can you elaborate.
Thanks.

Reply:
I think the comments should say y is Input activation. If you look at logistic_function's `Deriv() ` function, it calculates y*(1-y). 
So deriv function takes in input y, and returns f'(x) where y = f(x).

Reply:
I know y, x, f(x), f'(x) are very confusing I struggled a lot to get a hold of them.
But here's what I got.
The deriv function just takes an input (let's say x,y,x anything, here by convention its y) and gives output as the derivative of the input. Now the derivative is wrt the variable of which the input is a function.

> y = f(x);
> derivative = d(variable)/d(most basic input) = d(y)/d(x) = d(p)/d(x) = d(r)/d(x);

so what we need to do is the deriv function gets an input , it doesn't know anything about it and it just calculated its derivative and returns it.
This is what I think of the deriv function. Though I might be missing something.
Wanted to know if it's the same as you thought.
Thanks.

Reply:
Hi @prince776, ok first for a moment please forget the variable name in `Deriv ` function. If you are trying to express the derivative in terms of f(x) then you should provide the Inverse else it doesn't makes sense to express the derivatives in f(x). Why impose multiple computation when we can simply express the derivative in terms of x. And here i agree with @ojhalakshya, here in `Deriv` function y is merely a variable name by convention. And to be sure please check that the test are passing & look 1) tanh, logistic 2) swish,mish closely.

Reply:
Ohk sorry for jumping in this interesting conversation so I'm going to clarify something:
1. Derivatives are universal (independent of input). Most of the time piecewise functions can have derivatives expressed in piecewise functions.
2. What @prince776 has pointed out is, when we test the derivative we can utilize the information that the input has been passed through the `Fn` to improve simplicity. However that is not always the case, In case we need to call inverse and then perform derivative then we should just implement the derivative.
3.  What @bisakhmondal and @ojhalakshya have pointed out is same as point 1, we needn't worry about what the input is, we can simply implement the derivative (This is done incase the function is already a bit complex)

So a MWE is given below:

derivative of `tanh(x)` is `1 - tanh(x)*tanh(x)` However if I know that my input is already `tanh(x)` and I assume y = tanh(x) then I can write derivative as `1 - y*y`
However in reality the derivative will always `1 - tanh(x)*tanh(x)` and never `1-y*y`. These two would be equal only when y = `tanh(x)` and in testing we use this condition because when we calculate gradient we know that we have already passed it through input. So it doesn't really matter what you call the variable the important thing to remember is that whichever function we implement it can only have one derivative.
I hope this helps a bit. Thanks for all the work though, @prince776, @bisakhmondal, @ojhalakshya. 
I hope this is not completely vague and irrelevant. Looking forward to more great PRs from all of you. Thanks.